### PR TITLE
Fix rendering into shadow root

### PR DESCRIPTION
**Pull request by gaearon**
Fixes https://github.com/facebook/react/issues/11017.

1. Removes a unit test we always skip (it's from the times we ran tests in browsers).
2. Adds a fixture that actually verifies we can mount into shadow root.
3. Fixes mounting into shadow root.

Before:

screen shot 2017-10-02 at 19 53 57

After:

screen shot 2017-10-02 at 19 54 36

### CODE REVIEW

**Comment by aweary** [comment]
@gaearon do we need to update `validateDOMNesting` to check for `#fragment` too?

https://github.com/facebook/react/blob/master/src/renderers/dom/shared/validateDOMNesting.js#L290

**Comment by gaearon** [comment]
I don't think we need to. I intentionally changed it to skip that check since it's valid to render anything into a fragment.

**Comment by gaearon** [comment]
>LGTM, should we add a small test for mounting into a document fragment?

If you'd like as a follow up :-)

File: fixtures/dom/src/components/fixtures/custom-elements/index.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+return class MyElement extends HTMLElement {
+  constructor() {
+    super();
+    const shadowRoot = this.attachShadow({ mode:'open' });
+    ReactDOM.render(React.createElement(HelloWorld), shadowRoot);
+  }
+}`
+)(React, ReactDOM, HelloWorld);
+
+customElements.define('my-element', MyElement);
```
Comment (aweary) [inline fixtures/dom/src/components/fixtures/custom-elements/index.js]: @gaearon this is causing the fixtures to fail in browsers that don't have `customElements`. We should either provide a polyfill or only run this fixture in browsers that support the custom element registry.

**Pull request merged**